CMakeLists.txt

```diff
@@ -0,0 +1,193 @@
+cmake_minimum_required(VERSION 3.19 FATAL_ERROR)
```

Comment:
Is such a recent version of CMake really needed?

Reply:
Checked the reason, it's because I was using the [CheckLinkerFlag](https://cmake.org/cmake/help/v3.19/module/CheckLinkerFlag.html) and [CheckCompilerFlag](https://cmake.org/cmake/help/v3.19/module/CheckCompilerFlag.html) modules, which have been introduced in CMake 3.18 and 3.19, respectively.

I can resort to the older CheckCXXCompilerFlag and CheckCXXSourceCompiles modules with a bit more boilerplate code around them.

The oldest CMake version that can be supported without sacrificing more functionality - specifically the `add_link_options` command - is 3.14, released in February 2019.

As it's quite easy to install any recent CMake version and older versions have other bugs and issues,

Reply:
Ubuntu 20.04 [has cmake 3.16](https://packages.ubuntu.com/focal/cmake) so I think that would be a good target.